packages/amplify_datastore/android/src/main/kotlin/com/amazonaws/amplify/amplify_datastore/AmplifyDataStorePlugin.kt

```diff
@@ -15,37 +15,142 @@
+import com.amplifyframework.datastore.DataStoreItemChange
+import com.amplifyframework.datastore.appsync.SerializedModel
 import io.flutter.embedding.engine.plugins.FlutterPlugin
 import io.flutter.plugin.common.MethodCall
 import io.flutter.plugin.common.MethodChannel
 import io.flutter.plugin.common.MethodChannel.MethodCallHandler
 import io.flutter.plugin.common.MethodChannel.Result
-import io.flutter.plugin.common.PluginRegistry.Registrar
+import java.util.Date
+import java.util.UUID
 
 /** AmplifyDataStorePlugin */
-class AmplifyDataStorePlugin: FlutterPlugin, MethodCallHandler {
-  /// The MethodChannel that will the communication between Flutter and native Android
-  ///
-  /// This local reference serves to register the plugin with the Flutter Engine and unregister it
-  /// when the Flutter Engine is detached from the Activity
-  private lateinit var channel : MethodChannel
-
-  override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
-    channel = MethodChannel(flutterPluginBinding.binaryMessenger, "amplify_datastore")
-    channel.setMethodCallHandler(this)
-  }
-
-  override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
-    if (call.method == "getPlatformVersion") {
-      result.success("Android ${android.os.Build.VERSION.RELEASE}")
-    } else {
-      result.notImplemented()
+class AmplifyDataStorePlugin : FlutterPlugin, MethodCallHandler {
+
+    private lateinit var channel: MethodChannel
+    private val LOG = Amplify.Logging.forNamespace("amplify:flutter:datastore")
+
+    override fun onAttachedToEngine(
+            @NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
+        channel = MethodChannel(flutterPluginBinding.binaryMessenger,
+                                "com.amazonaws.amplify/datastore")
+        channel.setMethodCallHandler(this)
+        LOG.info("Added DataStore plugin")
+    }
+
+    override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
+        var data: HashMap<String, Any> = HashMap()
+        try {
+            data = checkArguments(call.arguments) as HashMap<String, Any>
+        } catch (e: Exception) {
+            prepareError(result, e, FlutterDataStoreFailureMessage.CASTING.toString())
+        }
+        when (call.method) {
+            "query" -> onQuery(result, data)
+            "configure" -> onConfigure(result, data)
+        }
+    }
+
+    private fun onConfigure(flutterResult: Result, request: HashMap<String, Any>) {
+        var modelSchemas: List<Map<String, Any>>? = null
+        if (request.containsKey("modelSchemas") && request["modelSchemas"] is List<*>) {
+            modelSchemas = request["modelSchemas"].safeCastToList()
+        } else {
+            prepareError(flutterResult,
+                         java.lang.Exception(FlutterDataStoreFailureMessage.MALFORMED.toString()),
+                         FlutterDataStoreFailureMessage.CASTING.toString())
```

Comment:
I think it would be valuable for prepareError to be passed localizedMessage or a recoverySuggestion for errors in the plugin layer. Eg, recovery suggestions around `modelSchema` or `modelName` (below in `onQuery`) not being the correct format and what they should be?

Reply:
These are not user errors, serialization and de-serialization errors happen 100% because of bugs in the library and a generic localizedError/recoverySuggestion should suffice. For instance, in your example `modelSchema` and `modelName` not being in the correct format would imply our conversion from Model to ModelSchema has errors.